scripts/linux/runE2ETest.sh

```diff
@@ -1073,6 +1079,7 @@ TRANSPORT_TYPE_4="${TRANSPORT_TYPE_4:-mqtt}"
 TWIN_UPDATE_FAILURE_THRESHOLD="${TWIN_UPDATE_FAILURE_THRESHOLD:-00:01:00}"
 METRICS_SCRAPE_FREQUENCY_IN_SECS="${METRICS_SCRAPE_FREQUENCY_IN_SECS:-300}"
 METRICS_UPLOAD_TARGET="${METRICS_UPLOAD_TARGET:-AzureLogAnalytics}"
+TEST_START_DELAY="${TEST_START_DELAY:-00:02:00}"
```

Comment:
Previously LH start right away and no start delay is required, why do we need to set it (for DMSender) in LH?

Reply:
Currently, dmsender will receive "internal server error" status code if it comes up before the receiver. I'm not sure why we are starting to see this now, but we have made some changes to the direct method modules since it was stable.

I don't see a problem using test start delay as opposed to a retry. Connectivity also uses the test start delay.

Reply:
It is ok to have test start delay.  I am concerned that the sequence number is increased if there is an exception.  In LH, the assumption is to increase sequence number if operation succeeds, otherwise retry with same sequence number.  If LH didn't get any result longer than 1s (not 100% sure the duration).  It will show a message in result like a warning.  If this behavior is changed, our LH test will fail if flaky network causes DM call failed once; then test analyzer will report missing records and failed test.  Correct me if my understanding is incorrect.

Reply:
Two things:
1. There is no sequence number for direct method tests. It sends status reports with pass/fail.
2. This problem only occurs at the beginning of longhaul. I think that having this problem should fail the tests if it occurs when both modules are online.

Reply:
1. Sequence number is in directMethodCount variable. https://github.com/Azure/iotedge/blob/c3efe9185bc2a7063748e0cc07ee3b806cdbc416/test/modules/DirectMethodSender/DirectMethodSenderBase.cs#L35.  And it reports in test result, https://github.com/Azure/iotedge/blob/c3efe9185bc2a7063748e0cc07ee3b806cdbc416/test/modules/DirectMethodReceiver/src/DirectMethodReceiver.cs#L52

However test analyzer generating LH DM report doesn't care about sequence number.

Last question: do we expect LH to fail due to DM call failed causing by flaky network drop within very short period?

Reply:
Ideally this shouldn't fail the test. I haven't seen a status code indicating network issues in the report. If we do see it, we should implement failure tolerance based on network exception type.